browser/android/youtube_script_injector/brave_youtube_script_injector_native_helper_browsertest.cc

```diff
@@ -0,0 +1,306 @@
+  // Wait for the resize to complete triggered by fullscreen change.
+  content::WaitForResizeComplete(web_contents());
+  // Assert the video is in fullscreen mode.
+  ASSERT_TRUE(
+      WaitForJsResult(web_contents(), "document.fullscreenElement !== null"));
+
+  youtube_script_injector::SetFullscreen(web_contents());
+
+  EXPECT_TRUE(WaitForJsResult(
+      web_contents(),
+      "document.querySelector('video.html5-main-video').paused === false"));
+}
+
+IN_PROC_BROWSER_TEST_F(BraveYouTubeScriptInjectorNativeHelperBrowserTest,
+                       NoOpIfPlayerIsNotFound) {
+  // Load a page without a video.
+  const GURL url = https_server_.GetURL("youtube.com", "/ytcftg_mock.html");
+  content::NavigateToURLBlockUntilNavigationsComplete(web_contents(), url, 1,
+                                                      true);
+
+  // Assert that movie player is not present.
+  ASSERT_TRUE(
+      content::EvalJs(web_contents(),
+                      "document.getElementById('movie_player') === null")
+          .ExtractBool());
+
+  std::string dom_before =
+      content::EvalJs(web_contents(), "document.body.innerHTML")
+          .ExtractString();
+  // Attempt to set fullscreen, which should not change anything.
+  youtube_script_injector::SetFullscreen(web_contents());
+
+  std::string dom_after =
+      content::EvalJs(web_contents(), "document.body.innerHTML")
+          .ExtractString();
+  // Assert that the DOM remains unchanged.
+  EXPECT_EQ(dom_before, dom_after);
+}
+
+IN_PROC_BROWSER_TEST_F(BraveYouTubeScriptInjectorNativeHelperBrowserTest,
+                       SetFullscreenOnElementsLoadingDelayed) {
+  const GURL url =
+      https_server_.GetURL("youtube.com", "/yt_fullscreen_delayed.html");
+  content::NavigateToURLBlockUntilNavigationsComplete(web_contents(), url, 1,
+                                                      true);
+
+  // Simluate delayed script load by presenting a player ('movie_player') that
+  // is not immediately available. Assert that movie player is present.
+  ASSERT_TRUE(
+      content::EvalJs(web_contents(),
+                      "document.getElementById('movie_player') !== null")
+          .ExtractBool());
+
+  // Assert that the video element (inside the player) is not present initially.
+  ASSERT_TRUE(content::EvalJs(
+                  web_contents(),
+                  "document.querySelector('video.html5-main-video') === null")
+                  .ExtractBool());
+  // Assert that the button (inside the player) to trigger fullscreen is not
+  // present initially.
+  ASSERT_TRUE(content::EvalJs(
+                  web_contents(),
+                  "document.querySelector('button.fullscreen-icon') === null")
+                  .ExtractBool());
+
+  youtube_script_injector::SetFullscreen(web_contents());
+
+  // Inject a script to simulate delayed loading of the video element fullscreen
+  // button.
+  InjectScript(u"window.simulateDelayedScriptLoad();");
```

Comment:
it would be better to have the source of the script in cpp so it's easier to see what is happening here

Reply:
Moved the content of the script directly in CPP file 👉 [`52acd4f` (#29431)](https://github.com/brave/brave-core/pull/29431/commits/52acd4f3ef5902623b3dc00e53f056d46554adc4)